### PR TITLE
fix(tmux_cmd): use shutil.which and only PATH to discover tmux

### PR DESCRIPTION
**Pull request by rockandska**
Follow-up of https://github.com/tmux-python/libtmux/issues/396#issuecomment-1221582103

Copy @mdeguzis

### CODE REVIEW

File: libtmux/common.py [hunk 1]
```diff
@@ -13,6 +13,7 @@
 import typing as t
 from distutils.version import LooseVersion
 from typing import Dict, Generic, KeysView, List, Optional, TypeVar, Union, overload
+from shutil import which as which
```
Comment (tony) [inline libtmux/common.py:16]: Nitpick: Fine to `import shutil` and use `shutil.which()`

File: tests/test_common.py [hunk 2]
```diff
@@ -174,16 +174,13 @@ def test_has_lte_version() -> None:
...
-
-
 def test_tmux_cmd_raises_on_not_found() -> None:
+    previous_path = os.environ["PATH"]
+    os.environ["PATH"] = ""
     with pytest.raises(TmuxCommandNotFound):
-        tmux_cmd("-V", tmux_search_paths=[], append_env_path=False)
+        tmux_cmd("-V")
+
+    os.environ["PATH"] = previous_path
```
Comment (tony) [inline tests/test_common.py]: Line 177-183:

[pytest](https://docs.pytest.org/)'s [`MonkeyPatch.setenv()`](https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest.MonkeyPatch.setenv)

```python
def test_tmux_cmd_raises_on_not_found(monkeypatch: pytest.MonkeyPatch) -> None:
    monkeypatch.setenv("PATH", "")
    with pytest.raises(TmuxCommandNotFound):
        tmux_cmd("-V")
```

Pushed a change for this

File: libtmux/common.py (hunk 1 above)
Comment (tony) [reply libtmux/common.py:16]: Looks good!

File: tests/test_common.py (hunk 2 above)
Comment (rockandska) [reply tests/test_common.py]: Sorry missed it :/
Let me apply it back

**Comment by tony** [comment]
@rockandska Do my commits on top look okay to you? Let me know if you have any feedback

File: libtmux/common.py (hunk 1 above)
Comment (rockandska) [reply libtmux/common.py:16]: Was reverted by one of us during cross force push

I let you remove it

**Comment by rockandska** [comment]
> @rockandska Do my commits on top look okay to you? Let me know if you have any feedback

The `from shutil import which` was pushed [back](https://github.com/tmux-python/libtmux/pull/407/files#diff-92e2d8e319de280e29a56e0514223e00c5b3f56f50eb7eea8caa5faf0673def1R16) during force push, I let you removed it

And not sure if it is relevant, but I didn't removed kwargs from `tmux_cmd` even if it is not used anymore

**Comment by tony** [comment]
> The from shutil import which was pushed [back](https://github.com/tmux-python/libtmux/pull/407/files#diff-92e2d8e319de280e29a56e0514223e00c5b3f56f50eb7eea8caa5faf0673def1R16) during force push, I let you remove it

Fixed

> And not sure if it is relevant, but I didn't removed kwargs from tmux_cmd even if it is not used anymore

I don't see this.

**Comment by tony** [comment]
@rockandska @mdeguzis

v0.15.0a1: [GitHub release](https://github.com/tmux-python/libtmux/releases/tag/v0.15.0a1), [git tag](https://github.com/tmux-python/libtmux/tree/v0.15.0a1), [PyPI package](https://pypi.org/project/libtmux/0.15.0a1/)

```console
pip install libtmux==0.15.0a1
```

**Comment by tony** [comment]
@rockandska Thank you for this!  Well done

Much cleaner + standard library + more explicit and predictable

**Comment by tony** [comment]
Live in v0.15.0

**Pull request merged**